indispensable-cosef/src/commonMain/kotlin/at/asitplus/signum/indispensable/cosef/CoseHeader.kt

```diff
@@ -131,6 +132,18 @@ data class CoseHeader(
     @SerialName("typ")
     val type: String? = null,
 ) {
+    /**
+     * Tries to compute a public key in descending order from [coseKey], [kid] or
+     * [certificateChain], and takes the first success or null.
+     */
+    val publicKey: CoseKey?
+        get() = coseKey?.let { CoseKey.deserialize(it).getOrNull() }
+            ?: kid?.let { CoseKey.fromDid(it.decodeToString()) }?.getOrNull()
+            ?: certificateChain?.let {
+                runCatching {
+                    X509Certificate.decodeFromDer(it)
+                }.getOrNull()?.publicKey?.toCoseKey()?.getOrThrow()
```

Comment:
do we want to ignore fields that are present but throw an exception when parsing? this feels like a design decision, but IMO if a coseKey field is present but invalid this should be an error, not silently ignored. @JesusMcCloud?

Reply:
agreed! null is koscher but an invalid key should result in an error